backend/handlers/handlers.go

```diff
@@ -294,15 +295,26 @@ func (handlers *Handlers) postAddAccountHandler(r *http.Request) (interface{}, e
 	keypath := signing.NewEmptyAbsoluteKeypath()
 	extendedPublicKey, err := hdkeychain.NewKeyFromString(jsonExtendedPublicKey)
 	if err != nil {
-		return nil, err
+		return map[string]interface{}{"success": false, "errorCode": "xpubInvalid"}, nil
+	}
+	if extendedPublicKey.IsPrivate() {
+		return map[string]interface{}{"success": false, "errorCode": "xpubInvalid"}, nil
```

Comment:
Since it's quick to change and very important information for the user, you could/should introduce an additional error code `xpubPrivate` with a corresponding message to warn the user that he/she should keep the xpriv confidential. (And after warning the user, we could still import the corresponding xpub as watch-only… 🤷‍♂️)